packages/cli/src/lib/test-env/client-config.ts

```diff
@@ -29,14 +27,21 @@ export function getTestEnvClientConfig(): Partial<BuilderConfig> {
 
   return {
     envs: {
-      [defaultPackages.ipfsResolver]: {
+      [DefaultBundle.embeds.asyncIpfsResolver.uri.uri]: {
```

Comment:
ultra nit: i am not sure about having `asyncIpfsResolver`, I feel we should just have `ipfsResolver`, the `async` is an implementation detail that the client config shouldn't need to be aware of?

Reply:
The IPFS resolver implementations can be found here: https://github.com/polywrap/uri-resolver-extensions/tree/master/implementations/ipfs

While I agree in principle that async should use sync resolver under-the-hood, it's easier to just build them as 2 separate wrappers as it helps conserve memory.

I agree with simplifying the naming @cbrzn so happy to make this change.

Reply:
The async IFPS resolver uses the IPFS HTTP client wrapper. The IPFS resolver wrapper implements the URI resolver interface, while the IPFS HTTP client is a more general IPFS HTTP client utility. 

This is somewhat necessary due to the way our concurrency feature works. The concurrency interface and plugin are designed for concurrent invocations. That implies there must be two wrappers: one that implements the Uri Resolver interface and another that is called with concurrency. The initial call to the URI Resolver cannot be parallel, because it must work the same way that other URI resolvers do. The call to the IPFS `cat` function is what needs to be parallelized. 

The reason it has `async` in the name is because there is a synchronous IPFS resolver that was made for use in clients that don't yet have concurrency plugins available. We can absolutely change the name in the default config. That makes sense to me!

Reply:
Also we could certainly implement different ways of parallelizing the ipfs resolution, but would leave this for the future on an as-need basis.

Reply:
benefit of using wrappers is they're super easy to swap out, just change the URI :D

Reply:
The IPFS resolver wrapper could call itself recursively, and it could all be done with one wrapper. But it would require a hard-coded URI in the wrapper. Let me know what you think!

Reply:
> The IPFS resolver wrapper could call itself recursively, and it could all be done with one wrapper. But it would require a hard-coded URI in the wrapper. Let me know what you think!

This has been a feature that's been requested a lot, allowing wrappers to know about their URI. IIRC @Niraj-Kamdar @nerfZael had a plan for how to implement this in wrap 0.2?